src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/AttendanceResourceIntTest.kt

```diff
@@ -91,18 +86,20 @@ class AttendanceResourceIntTest : IntegrationTestBase() {
     @BeforeEach
     fun setUp() {
       testData(repository) {
+        offender = offender(nomsId = "A1234AR") {
+          booking(agencyLocationId = "LEI")
+        }
+        offenderBooking = offender.latestBooking()
         programService {
           courseActivity = courseActivity {
             payRate()
             courseSchedule()
             courseScheduleRule()
           }
+          courseSchedule = courseActivity.courseSchedules.first()
         }
+        allocation = courseAllocation(offenderBooking, courseActivity)
```

Comment:
Most tests use an allocation so create on in the set up. Any tests needing a different or no allocation can set up from scratch.